flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/hybrid/HsResultPartitionReadSchedulerTest.java

```diff
@@ -0,0 +1,403 @@
+        readScheduler =
+                new HsResultPartitionReadScheduler(
+                        bufferPool,
+                        ioExecutor,
+                        new HsFileDataIndexImpl(NUM_SUBPARTITIONS),
+                        dataFilePath,
+                        HybridShuffleConfiguration.createConfiguration(
+                                bufferPool.getNumBuffersPerRequest(), NUM_SUBPARTITIONS));
+        subpartitionViewOperation = new TestingSubpartitionViewInternalOperation();
+    }
+
+    @AfterEach
+    void after() throws Exception {
+        bufferPool.destroy();
+        if (dataFileChannel != null) {
+            dataFileChannel.close();
+        }
+    }
+
+    // ----------------------- test run and register ---------------------------------------
+
+    @Test
+    void testRegisterReaderTriggerRun() throws Exception {
+        TestingHsSubpartitionFileReader.Factory factory =
+                new TestingHsSubpartitionFileReader.Factory();
+
+        assertThat(ioExecutor.numQueuedRunnables()).isZero();
+
+        readScheduler.registerNewSubpartition(0, subpartitionViewOperation, factory);
+
+        assertThat(ioExecutor.numQueuedRunnables()).isEqualTo(1);
+    }
+
+    @Test
+    void testBufferReleasedTriggerRun() throws Exception {
+        TestingHsSubpartitionFileReader.Factory factory =
+                new TestingHsSubpartitionFileReader.Factory();
+        Set<MemorySegment> memorySegmentSets = new HashSet<>();
+        AtomicReference<BufferRecycler> recycleRef = new AtomicReference<>(null);
+        factory.setReadBuffersConsumer(
+                (requestedBuffer, recycle) -> {
+                    while (!requestedBuffer.isEmpty()) {
+                        memorySegmentSets.add(requestedBuffer.poll());
+                    }
+                    recycleRef.set(recycle);
+                });
+
+        readScheduler.registerNewSubpartition(0, subpartitionViewOperation, factory);
+
+        ioExecutor.trigger();
+
+        assertThat(recycleRef).isNotNull();
+        assertThat(memorySegmentSets).hasSize(2);
+        assertThat(bufferPool.getAvailableBuffers()).isZero();
+
+        recycleRef.get().recycle(memorySegmentSets.iterator().next());
+
+        assertThat(ioExecutor.numQueuedRunnables()).isEqualTo(1);
+    }
+
+    /** Test all not used buffers will be released after run method finish. */
+    @Test
+    void testRunReleaseAllBuffers() throws Exception {
+        TestingHsSubpartitionFileReader.Factory factory =
+                new TestingHsSubpartitionFileReader.Factory();
+        AtomicBoolean prepareForSchedulingFinished = new AtomicBoolean(false);
+        factory.setPrepareForSchedulingRunnable(() -> prepareForSchedulingFinished.set(true));
+        factory.setReadBuffersConsumer(
+                (buffers, recycle) -> {
+                    assertThat(prepareForSchedulingFinished).isTrue();
+                    assertThat(buffers).hasSize(2);
```

Comment:
It would be better to have a constant `BUFFER_POOL_SIZE`, and use it like `hasSize(BUFFER_POOL_SIZE)` in the test cases.